pyo3-build-config/src/impl_.rs

```diff
@@ -440,6 +440,25 @@ print("mingw", get_platform().startswith("mingw"))
         let version = version.ok_or("missing value for version")?;
         let implementation = implementation.unwrap_or(PythonImplementation::CPython);
         let abi3 = abi3.unwrap_or(false);
+        // Fixup lib_name if it's not set
+        let lib_name = lib_name.or_else(|| {
+            if let Ok(Ok(target)) = env::var("TARGET").map(|target| target.parse::<Triple>()) {
+                if target.operating_system == OperatingSystem::Windows {
```

Comment:
If this logic is the same as the one linked in the cover letter, would it make sense to factor it out into a helper function?

Reply:
Done.